src/main/java/com/bfa/core/page/GooglePage.java

```diff
@@ -0,0 +1,75 @@
+package com.bfa.core.page;
+
+import com.bfa.core.BrowserType;
+import com.bfa.core.util.BrowserUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+
+/**
+ * Created by ninhdoan on 12/24/16.
+ */
+public class GooglePage extends GenericPage {
+
+   private static final String GOOGLE_URL = "https://google.com";
+
+   private static final String SEARCH_BOX_NAME = "q";
+
+   private static final String SEARCH_RESULT_ID = "resultStats";
+
+   private static final String RESULT_LINK_XPATH = "//*[@id='rso']//h3/a";
+
+   private static final long DEFAULT_WAIT_TIME = 5;
+
+
+   private WebElement searchBox;
+
+   private GooglePage(WebDriver webDriver) {
+      super(webDriver);
+   }
+
+   public static GooglePage openGoogle(BrowserType browserType) {
+      WebDriver webDriver = BrowserUtils.openBrowser(browserType);
+      GooglePage googlePage = new GooglePage(webDriver);
+      googlePage.navigateToGoogleUrl();
+      return googlePage;
+   }
+
+   private void navigateToGoogleUrl() {
+      webDriver.navigate().to(GOOGLE_URL);
+   }
+
+   public GooglePage search(String keyword) {
+      if (searchBox == null) {
+         waitForSearchBoxToBeReady(DEFAULT_WAIT_TIME);
+      }
+      searchBox.sendKeys(keyword);
+      searchBox.submit();
+      return this;
+   }
+
+   private void waitForSearchBoxToBeReady(long waitTimeInSeconds) {
+      WebDriverWait webDriverWait = new WebDriverWait(this.webDriver, waitTimeInSeconds);
+      searchBox = webDriverWait.until(ExpectedConditions.presenceOfElementLocated(By.name(SEARCH_BOX_NAME)));
+   }
+
+
+   public AnyPage openFirstResult() {
+      waitForSearchResultToBeReady(DEFAULT_WAIT_TIME);
+      List<WebElement> webElements = webDriver.findElements(By.xpath(RESULT_LINK_XPATH));
+      WebElement firstResult = webElements.get(0);
+      firstResult.click();
+      return new AnyPage(webDriver);
+   }
+
+   private void waitForSearchResultToBeReady(long waitTimeInSeconds) {
+      WebDriverWait webDriverWait = new WebDriverWait(this.webDriver, waitTimeInSeconds);
+      webDriverWait.until(ExpectedConditions.presenceOfElementLocated(By.id(SEARCH_RESULT_ID)));
```

Comment:
* Same here. Please update to use ```visibilityOfElementLocated```.
* Please also change the approach to use different locator type, not only ID.